### PR TITLE
Future

### PR DESCRIPTION
**Pull request by LanlanWU**

### CODE REVIEW

**Comment by LanlanWU** [comment]
Save the "save def" and "fetch function"

File: app.py [hunk 1]
```diff
@@ -118,6 +118,25 @@ def saverecipes():
...
+            access_token = get_access_token()
+            get_song_data(recipes_id, access_token)
+            valid_ids.add(recipes_id)
+        except KeyError:
+            pass
+
+    username = current_user.username
+    update_db_ids_for_user(username, valid_ids)
+
+    response = {"artist_ids": [a for a in recipes_id if a in valid_ids]}
```
Comment (cedwards57) [inline app.py:137]: artist ids?

File: app.py [hunk 2]
```diff
@@ -118,6 +118,25 @@ def saverecipes():
...
+    for recipes_id in recipes_id:
+        try:
+            access_token = get_access_token()
+            get_song_data(recipes_id, access_token)
+            valid_ids.add(recipes_id)
+        except KeyError:
+            pass
+
+    username = current_user.username
+    update_db_ids_for_user(username, valid_ids)
```
Comment (cedwards57) [inline app.py:135]: what is this function?

File: src/App.js [hunk 3]
```diff
@@ -1,80 +1,84 @@
...
-    const [mealData, setMealData] = useState(null)
+
+function App() {
+  const [query, setQuery] = useState("");
+  const [recipes, setRecipes] = useState(arges.recipes_id);
+  const [alert, setAlert] = useState("");
+ 
+
+  const APP_ID = "1bff948a";
+  const APP_KEY = "6684732180f3eec5d8f961b7abd3d412";
```
Comment (cedwards57) [inline src/App.js:18]: don't commit keys!

File: app.py [hunk 4]
```diff
@@ -118,6 +118,25 @@ def saverecipes():
...
 
+
+@app.route("/save", methods=["POST"])
+def save():
+   recipes_id = flask.request.json.get("recipes_id")
+    valid_ids = set()
+    for recipes_id in recipes_id:
+        try:
+            access_token = get_access_token()
+            get_song_data(recipes_id, access_token)
```
Comment (cedwards57) [inline app.py:129]: song data?

**Review by cedwards57 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**